src/components/checkbox/index.tsx

```diff
@@ -140,11 +161,15 @@ class Checkbox extends Component<CheckboxProps, CheckboxState> {
     };
   }
 
-  componentDidUpdate(prevProps: CheckboxProps) {
-    const {value} = this.props;
+  componentDidUpdate(prevProps: CheckboxProps, prevState: CheckboxState) {
+    const {value, onChangeValidity} = this.props;
     if (prevProps.value !== value) {
       this.animateCheckbox(value);
     }
+    
+    if (this.state.validationState && prevState.isValid !== this.state.isValid) {
```

Comment:
Why do we not notify the user before `validate` was pressed?
Does it affect the `Form`?

Reply:
I think this should be treated as a different mechanism, the user can `validate` or they can track the validity state (like we do in the `FormScreen`)

Reply:
So you're saying once the user passed 'required' the onValidityChange' should be called, no matter if the user invoked 'validate' or not. Right?

Reply:
Changed to call if required